ca_code/scripts/run_gen_texmean.py

```diff
@@ -0,0 +1,121 @@
+import logging
+import os
+import sys
+import cv2
+import torch as th
+from tqdm import tqdm
+from typing import List
+from torch.utils.data import DataLoader, Dataset
+from omegaconf import DictConfig, OmegaConf
+from addict import Dict as AttrDict
+
+from ca_code.utils.torchutils import to_device
+from ca_code.utils.render_drtk import RenderLayer
+from ca_code.utils.dataloader import BodyDataset, collate_fn
+from ca_code.utils.tex import get_tex_rl
+from ca_code.utils.geom import GeometryModule, index_image_impaint, make_uv_vert_index, make_uv_barys
+from ca_code.utils.lbs import LBSModule
+from ca_code.utils.tex import get_tex_rl
+
+logging.basicConfig(
+    format="[%(asctime)s][%(levelname)s][%(name)s]:%(message)s",
+    level=logging.INFO,
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+
+logger = logging.getLogger(__name__)
+
+def main(config: DictConfig):
+    device = th.device(f"cuda:0")
+    # force to get fully lit frames for tex mean extraction
+    config.data.fully_lit_only = True
+    config.data.partially_lit_only = False
+    train_dataset = BodyDataset(**config.data)
+    assets = AttrDict(train_dataset.static_assets)
+    geo_fn = GeometryModule(
+        th.LongTensor(assets.topology.vi),
+        assets.topology.vt,
+        assets.topology.vti,
+        assets.topology.v2uv,
+        uv_size=1024,
+        impaint=True,
+    ).to(device)
+    lbs_fn = LBSModule(
+        assets.lbs_model_json,
+        assets.lbs_config_dict,
+        assets.template_mesh_unscaled[None],
+        assets.skeleton_scales,
+        global_scaling=[10.0, 10.0, 10.0],  # meter
+    ).to(device)
+
+    rl = RenderLayer(
+        h=config.model.renderer.image_height,
+        w=config.model.renderer.image_width,
+        vt=geo_fn.vt,
+        vi=geo_fn.vi.int(),
+        vti=geo_fn.vti,
+        flip_uvs=False,
+    ).to(device)
+
+    uv_size = (1024, 1024)
+    inpaint_threshold = 100.0
+    vt = geo_fn.vt
+    vi = geo_fn.vi
+    vti = geo_fn.vti
+    index_image = make_uv_vert_index(
+        vt, vi, vti, uv_shape=uv_size, flip_uv=True
+    ).cpu()
+    face_index, bary_image = make_uv_barys(vt, vti, uv_shape=uv_size, flip_uv=True)
+
+    # inpaint index uv images
+    index_image, bary_image = index_image_impaint(
+                    index_image, bary_image, inpaint_threshold
+                )
```

Comment:
nit: strange indentation in closing parenthesis

Reply:
Updated.